### ISSUE TITLE
Keyboard navigation on Dropdown

### ISSUE THREAD

**Issue by d-lane** [angular-ui/bootstrap#5206]
When navigating a ui.bootstrap.dropdown element with keyboard-nav enabled the focused element forgets where it is meant to be if you navigate with tabs first and then arrow keys. For instance, if you open the dropdown entitled 'Dropdown with keyboard navigation' on the [demo page](https://angular-ui.github.io/bootstrap/) and tab to the final element, then press the down key the focus moves as if the first item was focused (as if tab hadn't been pressed).

**Comment by icfantv** [comment]
I think it's actually wrapping.  If you don't wait until the last element, it's working as designed.

**Comment by wesleycho** [comment]
Is this a bug @icfantv? Seems like expected behavior currently.

**Comment by icfantv** [comment]
@wesleycho, IIRC, it's not supposed to be wrapping.  If one tabs to an item before the last one and then use the arrow keys, it won't wrap when one reaches the bottom of the dropdown.

**Comment by wesleycho** [comment]
If the concern is with accessibility, it doesn't seem to say anything about what dropdowns should do. [This example](http://staff.washington.edu/tft/tests/menus/simplyaccessible/index.html) shows one where it wraps, but that isn't an indication that this is definitively the correct way. That dropdown's behavior does seem to agree with ours though.

**Comment by wesleycho** [comment]
So I took another quick look at this, and there is one example that supports the behavior suggested in the original post [here](http://terrillthompson.com/tests/menus/accessible-mega-menu/test.html), but the link I posted in the comment prior suggests that our current implementation is correct. Interestingly enough, it appears that the same person might have been involved in both versions, as my prior link mentions his page and links to the same staff.washington.edu domain and his homepage mentions his employment by the University of Washington.

Seeing this, and that there is no ironclad standard here, I am going to close this as won't change, as our dropdown seems to meet the requirements and that there is no universal approach when it comes to dropdowns.
